File: FSM/Entity/EngineerVisitAssetWorkSheets/EngineerVisitAssetWorkSheetValidatorASHP.cs
// Decompiled with JetBrains decompiler
// Type: FSM.Entity.EngineerVisitAssetWorkSheets.EngineerVisitAssetWorkSheetValidatorASHP
// Assembly: FSM, Version=1.12.1.0, Culture=neutral, PublicKeyToken=null
// MVID: FAB6F0E2-B68A-47AB-AF95-A9E31DED7575
// Assembly location: C:\github\gabriel.fieldservicemanager\src\bin\x86\Debug\FSM.exe

using Microsoft.VisualBasic.CompilerServices;
using System.Collections;

namespace FSM.Entity.EngineerVisitAssetWorkSheets
{
  public class EngineerVisitAssetWorkSheetValidatorASHP : BaseValidator
  {
    public void Validate(
      EngineerVisitAssetWorkSheet oEngineerVisitAssetWorkSheet)
    {
      if (oEngineerVisitAssetWorkSheet.Errors.Count > 0)
      {
        foreach (object error in oEngineerVisitAssetWorkSheet.Errors)
          this.AddCriticalMessage(Conversions.ToString((error != null ? (DictionaryEntry) error : new DictionaryEntry()).Value));
      }
      if (oEngineerVisitAssetWorkSheet.AssetID == 0)
        this.AddCriticalMessage("Appliance Missing");
      if (oEngineerVisitAssetWorkSheet.LandlordsApplianceID == 0)
        this.AddCriticalMessage("Landlords Appliance Missing");
      if (oEngineerVisitAssetWorkSheet.FlueFlowTestID == 0)
        this.AddCriticalMessage("Pressure Missing");
      if (oEngineerVisitAssetWorkSheet.SpillageTestID == 0)
        this.AddCriticalMessage("Operation Missing");
      if (oEngineerVisitAssetWorkSheet.FlueTerminationSatisfactoryID == 0)
        this.AddCriticalMessage("Safety devices Missing");
      if (oEngineerVisitAssetWorkSheet.VisualConditionOfFlueSatisfactoryID == 0)
        this.AddCriticalMessage("Stability Missing");
      if (oEngineerVisitAssetWorkSheet.VentilationProvisionSatisfactoryID == 0)
        this.AddCriticalMessage("Free air Missing");
      if (oEngineerVisitAssetWorkSheet.SafetyDeviceOperationID == 0)
        this.AddCriticalMessage("Safety device operation Missing");
      if (oEngineerVisitAssetWorkSheet.SweepOutcomeID == 0)
        this.AddCriticalMessage("Legionella Missing");
      if (oEngineerVisitAssetWorkSheet.OverallID == 0)
        this.AddCriticalMessage("Overall condition Missing");
      if (oEngineerVisitAssetWorkSheet.DischargeID == 0)
        this.AddCriticalMessage("Glycol Missing");
      if (oEngineerVisitAssetWorkSheet.ApplianceServiceOrInspectedID == 0)
        this.AddCriticalMessage("Appliance service Missing");
      if (oEngineerVisitAssetWorkSheet.ApplianceSafeID == 0)
        this.AddCriticalMessage("Appliance Safety Missing");
      if (this.ValidatorMessages.CriticalMessages.Count > 0)
        throw new ValidationException((BaseValidator) this);
    }
  }
}

File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitAssetWorkSheet/EngineerVisitAssetWorkSheetValidatorASHP.vb
Namespace Entity

Namespace EngineerVisitAssetWorkSheets
        Public Class EngineerVisitAssetWorkSheetValidatorASHP
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oEngineerVisitAssetWorkSheet As EngineerVisitAssetWorkSheet)

                'make sure that contact object is valid
                If oEngineerVisitAssetWorkSheet.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oEngineerVisitAssetWorkSheet.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If

                If oEngineerVisitAssetWorkSheet.AssetID = Nothing Then
                    Me.AddCriticalMessage("Appliance Missing")
                End If

                If oEngineerVisitAssetWorkSheet.LandlordsApplianceID = Nothing Then
                    Me.AddCriticalMessage("Landlords Appliance Missing")
                End If

                If oEngineerVisitAssetWorkSheet.FlueFlowTestID = Nothing Then
                    Me.AddCriticalMessage("Pressure Missing")
                End If

                If oEngineerVisitAssetWorkSheet.SpillageTestID = Nothing Then
                    Me.AddCriticalMessage("Operation Missing")
                End If

                If oEngineerVisitAssetWorkSheet.FlueTerminationSatisfactoryID = Nothing Then
                    Me.AddCriticalMessage("Safety devices Missing")
                End If

                If oEngineerVisitAssetWorkSheet.VisualConditionOfFlueSatisfactoryID = Nothing Then
                    Me.AddCriticalMessage("Stability Missing")
                End If

                If oEngineerVisitAssetWorkSheet.VentilationProvisionSatisfactoryID = Nothing Then
                    Me.AddCriticalMessage("Free air Missing")
                End If

                If oEngineerVisitAssetWorkSheet.SafetyDeviceOperationID = Nothing Then
                    Me.AddCriticalMessage("Safety device operation Missing")
                End If

                If oEngineerVisitAssetWorkSheet.SweepOutcomeID = Nothing Then
                    Me.AddCriticalMessage("Legionella Missing")
                End If

                If oEngineerVisitAssetWorkSheet.OverallID = Nothing Then
                    Me.AddCriticalMessage("Overall condition Missing")
                End If

                If oEngineerVisitAssetWorkSheet.DischargeID = Nothing Then
                    Me.AddCriticalMessage("Glycol Missing")
                End If

                If oEngineerVisitAssetWorkSheet.ApplianceServiceOrInspectedID = Nothing Then
                    Me.AddCriticalMessage("Appliance service Missing")
                End If

                If oEngineerVisitAssetWorkSheet.ApplianceSafeID = Nothing Then
                    Me.AddCriticalMessage("Appliance Safety Missing")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace